app.go

```diff
@@ -480,10 +480,6 @@ func New(opts ...Option) *App {
 		m.build(app, app.container)
 	}
 
-	for _, m := range app.modules {
-		m.provideAll()
-	}
-
 	frames := fxreflect.CallerStack(0, 0) // include New in the stack for default Provides
```

Comment:
Let's add a comment here describing why the Fx-provided dependencies are provided to the scope prior to the user provided types.